ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-security.j2

```diff
@@ -110,4 +110,28 @@ management security
       crl {{ crl }}
 {%         endfor %}
 {%     endfor %}
+{%     if management_security.shared_secret_profiles is arista.avd.defined %}
+{%         for profile in management_security.shared_secret_profiles | arista.avd.natural_sort("profile_name") %}
+   session shared-secret profile {{ profile.profile_name }}
+{%             if profile.secrets is arista.avd.defined %}
+{%                 for secret in profile.secrets | arista.avd.natural_sort("name") %}
+{%                     if secret.name is arista.avd.defined and secret.secret is arista.avd.defined and secret.lifetime is arista.avd.defined %}
+{%                         if secret.lifetime.infinite is arista.avd.defined(true) %}
+      secret {{ secret.name }} {{ secret.secret | arista.avd.hide_passwords(hide_passwords) }} infinite
+{%                         endif %}
+{%                     elif secret.name is arista.avd.defined and secret.secret is arista.avd.defined and secret.receive_lifetime is arista.avd.defined and secret.transmit_lifetime is arista.avd.defined %}
```

Comment:
This logic seems duplicated. Please try to work out a better nesting.

Reply:
Seems like this whole section would benefit from a cli token approach instead. Where you create a temp variable and append cli options to it depending on the given inputs.

Reply:
Used temporary variable to avoid repeated words in template.